zebra-state/src/service/check.rs

```diff
@@ -1,19 +1,31 @@
 //! Consensus critical contextual checks
 
+use std::borrow::Borrow;
+
 use zebra_chain::{
     block::{self, Block},
     parameters::Network,
+    parameters::POW_AVERAGING_WINDOW,
+    work::difficulty::CompactDifficulty,
 };
 
 use crate::{PreparedBlock, ValidateContextError};
 
 use super::check;
 
+pub mod difficulty;
+use difficulty::{AdjustedDifficulty, POW_MEDIAN_BLOCK_SPAN};
```

Comment:
Since the ordering doesn't matter we should be consistent about always ordering use statements before mod statements.

```suggestion
use difficulty::{AdjustedDifficulty, POW_MEDIAN_BLOCK_SPAN};

pub mod difficulty;
```